src/field_theory/polynomial_galois_group.lean

```diff
@@ -47,15 +47,45 @@ begin
   rwa [eq_top_iff, ←splitting_field.adjoin_roots, algebra.adjoin_le_iff],
 end
 
-instance [h : fact (p.splits (ring_hom.id F))] : unique p.gal :=
+instance unique_gal_of_splits [h : fact (p.splits (ring_hom.id F))] : unique p.gal :=
 { default := 1,
   uniq := λ f, alg_equiv.ext (λ x, by { obtain ⟨y, rfl⟩ := algebra.mem_bot.mp
     ((subalgebra.ext_iff.mp ((is_splitting_field.splits_iff _ p).mp h) x).mp algebra.mem_top),
     rw [alg_equiv.commutes, alg_equiv.commutes] }) }
 
-instance : unique (0 : polynomial F).gal :=
+instance unique_gal_zero : unique (0 : polynomial F).gal :=
 begin
-  haveI : fact ((0 : polynomial F).splits (ring_hom.id F)) := splits_zero _,
+  haveI : fact ((0 : polynomial F).splits (ring_hom.id F)) := splits_zero (ring_hom.id F),
+  apply_instance,
+end
+
+instance unique_gal_one : unique (1 : polynomial F).gal :=
+begin
+  haveI : fact ((1 : polynomial F).splits (ring_hom.id F)) := splits_one (ring_hom.id F),
```

Comment:
All these `haveI` proofs suggest to me that it would be useful to have a lemma version of `unique_gal_of_splits`.